src/main/java/com/chainstep/atlas/example/CustomPolicy.java

```diff
@@ -5,27 +5,142 @@
+            String jsonString = ow.writeValueAsString(object);
+            //log.info("returning JSON String: " + jsonString);                      
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            
+            // Parse the JSON string into an array of JsonNodes
+            JsonNode rootNode = objectMapper.readTree(jsonString);
+            
+            // Access the first element in the array (the credential itself)
+            JsonNode credentialNode = rootNode.get(0);
+
+            // Check if the credential is signed by AtlasXAuth
+            if(isSignedByAtlasXAuth(credentialNode)){
+                log.info("isSignedByAtlasXAuth true");   
+            }
+            else{
+                log.info("isSignedByAtlasXAuth false! NO access!");
+                return VerificationPolicyResult.Companion.failure();  // Fail if not signed by AtlasXAuth
+            }
+
+            // Check if the credential has the required legal registration number
+            if(isFromLegalRegistrationNumber(credentialNode, legalRegistrationNumString)){
+                log.info("isFromLegalRegistrationNumber true");   
+            }
+            else{
+                log.info("isFromLegalRegistrationNumber false! NO access!");
+                return VerificationPolicyResult.Companion.failure();  // Fail if the legal registration number does not match
+            }
+
+            // Check if the credential has the required role
+            if(isRole(credentialNode, roleToGrantAccessString)){
+                log.info("isRole true");   
+            }
+            else{
+                log.info("isRole false! NO access!");
+                return VerificationPolicyResult.Companion.failure();  // Fail if the role is not "remoteControl"
+            }
+
+            // If all checks pass, return success
+            return VerificationPolicyResult.Companion.success();
+            
+        } catch (Exception e) {
+            log.info("error during parsing credential subject" + e);
+            return VerificationPolicyResult.Companion.failure(new Exception("parsing Error in the CustomPolicy"));
+        }
+
     }
+
+        private boolean isSignedByAtlasXAuth(JsonNode jsonNode ) {
+            // get to the legalRegistrationNumber field
+            JsonNode proofCreator = jsonNode.get("proof").get("verificationMethod");
+            log.info("proofCreator  : " + proofCreator.toString());
+            log.info("issuer  : " + jsonNode.get("issuer").asText());
+            
+            if (proofCreator.toString().contains("did:web:authority.atlas.cartrust.com:gx:did")){
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+
+    // Method to check if the credential's legal registration number matches the expected one
+    private boolean isFromLegalRegistrationNumber(JsonNode jsonNode, String legalRegistrationNumber) {
```

Comment:
The thing we can discuss here is that should we have multiple functions in a class for these checks or should we have a separate policy class?

Reply:
I think we will have the check for the company and the role together.
We are thinking about making them configurable, so that we can set parameters in the application.yaml for each offering.
But this will be done in the next step.